app/controllers/posts_controller.rb

```diff
@@ -23,14 +23,24 @@ def create
     respond_to do |format|
       format.html do
         if new_post.save
-          redirect_to post_path(params[:user_id]), notice: 'Post was successfully created.'
+          redirect_to user_posts_path(params[:user_id]), notice: 'Post was successfully created.'
         else
           render :new, alert: 'Error occurred, please try again. Post not saved'
         end
       end
     end
   end
 
```

Comment:
Kindly, use `CanCanCan` for Post authorization here in `Post Controller`. To do that you can include something like this at the top between lines 1 and 2:
```
load_and_authorize_resource
```

You can read more [here](https://www.rubydoc.info/github/CanCanCommunity/cancancan/CanCan%2FControllerAdditions%2FClassMethods:load_and_authorize_resource).

Reply:
Thanks for the suggestion @Mre55 . Will implement